src/main/java/org/candlepin/subscriptions/db/EventRecordRepository.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2020 Red Hat, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Red Hat trademarks are not licensed under GPLv3. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.subscriptions.db;
+
+import org.candlepin.subscriptions.db.model.EventRecord;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+
+import java.time.OffsetDateTime;
+import java.util.UUID;
+import java.util.stream.Stream;
+
+/**
+ * DB repository for Event records.
+ *
+ * @see EventRecord
+ * @see org.candlepin.subscriptions.json.Event
+ */
+public interface EventRecordRepository extends JpaRepository<EventRecord, UUID> {
+
+    /**
+     * Fetch a stream of events for a given account for a given time range.
+     *
+     * The events returned include those at begin and up to (but not including) end.
+     *
+     * @param accountNumber account number
+     * @param begin start of the time range (inclusive)
+     * @param end end of the time range (exclusive)
+     * @return Stream of EventRecords
+     */
+    Stream<EventRecord> findByAccountNumberAndTimestampGreaterThanEqualAndTimestampLessThanOrderByTimestamp(
```

Comment:
Can this be `findByAccountNumberAndTimestampBetween` or does it need to be explicit because of the inclusive/exclusive?

Reply:
@lindseyburnett it does need to be this mess because of the semantics imposed here.

Given a couple of events that represents whole hours, I'd expect them to look something like this:

| timestamp | event |
|--|--|
|2021-01-01T00:00:00Z|{"expiration":"2021-01-01T01:00:00Z,..."}
|2021-01-01T01:00:00Z|{"expiration":"2021-01-01T02:00:00Z,..."}

pseudo-sql:
```sql
select * from events where timestamp where timestamp between '2021-01-01T00:00:00Z' and '2021-01-01T01:00:00Z'
```
yields both events since SQL between is inclusive, so asking for all events in the hour is actually a bit more awkward with between... we'd have to do something like

```sql
select * from events where timestamp where timestamp between '2021-01-01T00:00:00Z' and '2021-01-01T00:59:59.999Z'
```

I personally dislike `between` for this use case for this reason.

I will leave some more explanation in the JavaDoc for this.

Reply:
Done